src/FormField/MultiLine.php

```diff
@@ -496,18 +520,23 @@ public function setModel(\atk4\data\Model $model, $fields = [], $modelRef = null
         $this->rowFields = array_merge([$m->id_field], $fields);
 
         foreach ($this->rowFields as $fieldName) {
+            $enumValues = null;
             $field = $m->getField($fieldName);
 
             if (!$field instanceof \atk4\data\Field) {
                 continue;
             }
-            $type = $field->type ? $field->type : 'string';
+
+            $type = $this->getFieldType($field);
+            if ($type === 'enum') {
+                $enumValues = $this->getFieldEnumValues($field);
+            }
 
             if (isset($field->ui['form'])) {
                 $type = $field->ui['form'][0];
```

Comment:
If I get it correctly, `type` tells the JS how to render the input (Line, DropDown, TextArea etc). As of now the values of `type` are a mix of atk4 Data Field Types (string, integer, text etc). and Atk UI FormField types (DropDown, Line etc).
As its about rendering UI inputs, maybe we should be consistent with Atk UI FormField Types and always pass only these values (Using Line as Default)?

Reply:
Agree with you that this is somehow confusing. See comment below for better approach.